src/writer.f90

```diff
@@ -80,7 +80,7 @@ subroutine write_detected_photons(detectors)
         end subroutine write_detected_photons
 
 
-        subroutine write_data(array, filename, state, dict, overwrite)
+        subroutine write_data(array, filename, dict, overwrite)
```

Comment:
Hi Jack, you seem to have accidentally deleted state from the input variables so the code won't compile